engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -1248,6 +1248,9 @@ public int hashCode()
   @Override
   public <T extends Operator> T addOperator(String name, T operator)
   {
+    if (name == null || name.isEmpty()) {
```

Comment:
Use `checkArgument()` (statically import `com.google.common.base.Preconditions.checkArgument`) and `isNullOrEmpty()` (statically import `com.google.common.base.Strings.isNullOrEmpty`).